src/App.tsx

```diff
@@ -3,9 +3,68 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
+import { getMovie } from './api';
+import { MovieData } from './types/MovieData';
+import { ResponseError } from './types/ReponseError';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState(false);
+  const [loader, setLoader] = useState(false);
+
+  const changeQuery = (text: string) => {
+    setError(false);
+    setQuery(text);
+  };
+
+  function instanceOfMovie(data: MovieData | ResponseError):
+    data is MovieData {
+    return 'imdbID' in data;
+  }
+
+  const getMovieFromServer = async (text: string) => {
+    setLoader(true);
+
+    try {
+      const isMovie = await getMovie(text);
+
+      if (instanceOfMovie(isMovie)) {
+        setMovie({
+          title: isMovie.Title,
+          description: isMovie.Plot,
+          imgUrl: isMovie.Poster === 'N/A'
+            ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+            : isMovie.Poster,
+          imdbUrl: `https://www.imdb.com/title/${isMovie.imdbID}`,
+          imdbId: isMovie.imdbID,
+        });
+      } else {
+        setError(true);
+      }
+    } finally {
+      setLoader(false);
+    }
```

Comment:
Super IDK why we had to use it. i had no chances in my solution to get perpetual loading.

Reply:
https://prnt.sc/ZjQ-KHuBo3ER it was on button only)